src/routing/approved.rs

```diff
@@ -1158,6 +1176,36 @@ impl Approved {
         let age = peer.age();
         let signature = proof.signature.clone();
 
+        // A vote of half-aged Left peer means such peer is trying to rejoin with the same name,
+        // and shall be relocated immediately. The flow will return early during the process of
+        // update_member, as `Left -> Relocated` is not allowed.
+        //
+        // TODO: shall we make `Left -> Relocated` to avoid the left node is being repeatedly used
+        //       to generate higher new nodes?
+        if let Some(info) = self.section.members().get(member_info.peer.name()) {
+            if info.peer.age() > age {
```

Comment:
Should we also check that `info.state == Relocated` ?

Reply:
don't need to.
as this is handling a concensused Vote::Offline, and there is logic added that a Relocated node is not allowed to trigger a Vote::Offline.
i.e. A relocated Node is not allowed to rejoin with its origin name